mesonbuild/compilers/compilers.py

```diff
@@ -487,6 +487,7 @@ def __init__(self, exelist: T.List[str], version: str,
                  linker: T.Optional['DynamicLinker'] = None,
                  full_version: T.Optional[str] = None, is_cross: bool = False):
         self.exelist = exelist
+        self.exelist_no_ccache = exelist[1:] if exelist and 'ccache' in exelist[0] else exelist
```

Comment:
You mentioned sccache in the description, but this won't work for sccache, right?

Reply:
Also, what if ccache is used as 'ccache.exe'?

Reply:
`'ccache' in exelist[0]` that catch any string that contains "ccache", that includes "sccache" and "ccache.exe". But that deserves a comment to make it explicit I guess.

Reply:
A bit worried that this might lead to confusing failures with CC containing the string ccache somewhere. Perhaps this should only happen when meson added [s]ccache itself, and then it should remember that?

Reply:
Ok, I have to make a confession, compilers/detect.py does have ccache separately, but it would require changing the constructor of eeeeevery compiler classes to take ccache separately to exelist. This PR is a cheap lazy trick :P

But turns out, a few clever sed commands can easily keep ccache separated from compiler exelist, so I changed this PR to do that.